tools/src/h5repack/h5repack.c

```diff
@@ -242,7 +242,7 @@ copy_named_datatype(hid_t type_in, hid_t fidout, named_dt_t **named_dt_head_p, t
         while (dt && token_cmp) {
             dt = dt->next;
 
-            if (H5Otoken_cmp(type_in, &dt->obj_token, &oinfo.token, &token_cmp) < 0)
+            if (dt && H5Otoken_cmp(type_in, &dt->obj_token, &oinfo.token, &token_cmp) < 0)
                 H5TOOLS_GOTO_ERROR(H5I_INVALID_HID, "failed to compare object tokens");
         }
 
```

Comment:
I think your fix is correct but the logic here is a bit convoluted. Lines 238 through 247 appear to boil down to a linked-list search:

```
/* Search the stack for the datatype. */
for (dt = *named_dt_head_p; dt != NULL; dt = dt->next) {
    if (H5Otoken_cmp(type_in, &dt->obj_token, &oinfo.token, &token_cmp) < 0)
        H5TOOLS_GOTO_ERROR(H5I_INVALID_HID, "failed to compare object tokens");

    if (token_cmp == 0)
        break; // found it!
}
```

What do you think?

Reply:
Indeed seems nice, will make that change...